src/ucs/datastruct/string_buffer.c

```diff
@@ -135,6 +135,20 @@ void ucs_string_buffer_append_flags(ucs_string_buffer_t *strb, uint64_t mask,
     ucs_string_buffer_rtrim(strb, ",|");
 }
 
+void ucs_string_buffer_append_list(ucs_string_buffer_t *strb, const char *sep,
+                                   const char **strs, size_t count)
+{
+    size_t i;
+
+    for (i = 0; i < count; ++i) {
+        if (i > 0) {
+            ucs_string_buffer_appendf(strb, "%s%s", sep, strs[i]);
+        } else {
+            ucs_string_buffer_appendf(strb, "%s", strs[i]);
+        }
+    }
```

Comment:
minor: we can remove branch in the loop:
```suggestion
    size_t i;
    if (count > 0) {
        ucs_string_buffer_appendf(strb, "%s", strs[0]);
    }
    for (i = 1; i < count; ++i) {
        ucs_string_buffer_appendf(strb, "%s%s", sep, strs[i]);
    }
```